fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1209,6 +1212,170 @@ public RecordCursor<IndexEntry> scanIndex(@Nonnull Index index, @Nonnull IndexSc
         return context.instrument(FDBStoreTimer.Events.SCAN_INDEX_KEYS, result);
     }
 
+    @Nonnull
+    @Override
+    public RecordCursor<FDBIndexedRecord<Message>> scanIndexPrefetch(@Nonnull Index index,
+                                                                     @Nonnull TupleRange range,
+                                                                     @Nonnull final KeyExpression commonPrimaryKey,
+                                                                     @Nullable byte[] continuation,
+                                                                     @Nonnull ScanProperties scanProperties,
+                                                                     @Nonnull final IndexOrphanBehavior orphanBehavior) {
+        return scanIndexPrefetchInternal(index, range, commonPrimaryKey, continuation, serializer, scanProperties, orphanBehavior);
+    }
+
+    @Nonnull
+    protected <M extends Message> RecordCursor<FDBIndexedRecord<M>> scanIndexPrefetchInternal(@Nonnull final Index index,
+                                                                                @Nonnull final TupleRange range,
```

Comment:
Given that the only supported `IndexScanBounds` implementer would be `IndexScanRange` and the only supported `IndexScanType` would be `BY_VALUE`, do you anticipate a check up front to reject the unsupported options?

Reply:
Yeah, I think for now, we'd need to check the `IndexScanBounds` to extract the `IndexScanRange` anyway, like is done in `IndexMaintainer::scan`: https://github.com/foundationdb/fdb-record-layer/blob/2eadf38f86b5f58f806852c40bea821135feeba5/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainer.java#L110

But I think taking an `IndexScanBounds` and doing the extraction is a bit more forward looking, though it may look a little silly now

Reply:
Done